src/stapi_fastapi/routers/root_router.py

```diff
@@ -140,21 +140,51 @@ def get_root(self, request: Request) -> RootResponse:
     def get_conformance(self, request: Request) -> Conformance:
         return Conformance(conforms_to=self.conformances)
 
-    def get_products(self, request: Request) -> ProductsCollection:
-        return ProductsCollection(
-            products=[pr.get_product(request) for pr in self.product_routers.values()],
-            links=[
+    def get_products(
+        self, request: Request, next: str | None = None, limit: int = 10
+    ) -> ProductsCollection:
+        start = 0
+        product_ids = [*self.product_routers.keys()]
+        try:
+            if next:
+                start = product_ids.index(next)
+        except ValueError:
+            logging.exception("An error occurred while retrieving orders")
+            raise NotFoundException(detail="Error finding pagination token") from None
+        end = min(start + limit, len(product_ids))
+        ids = product_ids[start:end]
+        links = [
+            Link(
+                href=str(request.url_for(f"{self.name}:list-products")),
+                rel="self",
+                type=TYPE_JSON,
+            ),
+        ]
+        if end < len(product_ids) and end != 0:
```

Comment:
I prefer this written as 

```suggestion
        if end > 0 and end < len(product_ids):
```